### DISCUSSION TITLE
Client and Server components questions?

### DISCUSSION THREAD

**Discussion by Djboy08** [vercel/next.js#44739] [App Router]
Excuse my ignorance as I am pretty new to next and I wanted to ask a few questions in hopes someone has answers before I test and trial myself.

Firstly, I noticed in docs they highly suggest client components as leaf nodes in the tree. Can someone give me a thorough explanation on why this should be the case? Aswell as if this is a highly reccomended thing, how would one manage to make a theme provider with that requires a client component to be all the way up in the root layout. 

Another one of my questions is regarding passing state from a client to server component. Now in react we do a lot of 'lifting state' up to parent components but when client components should be leaves on a node, if there are two components that need to be able to share state or change it how would it work if there was server components inbetween?

Here is a example:

```js
<RootLayout>
    <ServerComponent>
         <ClientComponent/>
         <ServerComponent>
             <ClientComponent/>
         </ServerComponent>
    </ServerComponent>
</RootLayout>
```

because from my understanding I would HAVE to wrap anything I wanted to share state in a client component, and then the question is asked if I can pass that state down from server to client components

```js
<RootLayout>
    <ServerComponent>
         <ClientComponent>
             <ClientComponent/>
             <ServerComponent>
                 <ClientComponent/>
             </ServerComponent>
         </ClientComponent>
    </ServerComponent>
</RootLayout>
```

**Comment by dapjeongneo** [comment]
did anyone figure this out? i have the same question in mind. does it mean client wraps around server?

**Comment by liamgsmith** [comment]
I'm trying to get my head around this as well; did anyone come up with a good way of thinking about it?
